src/ContractInteractor.ts

```diff
@@ -1,919 +1,601 @@
+      };
+    }
+
+    // First call the verifier
+    try {
+      await this._deployVerifier
+        .connect(relayWorker)
+        .verifyRelayedCall(relayRequest, signature);
+    } catch ({ message }) {
+      return {
+        verifierAccepted: false,
+        reverted: false,
+        returnValue: `view call to 'deploy call' reverted in verifier: ${
+          message as string
+        }`,
+      };
+    }
+
+    // If the verified passed, try relaying the transaction (in local view call)
+    try {
+      const res = await relayHub
+        .connect(relayWorker)
+        .deployCall(relayRequest, signature, {
+          gasPrice: relayData.gasPrice,
+          gasLimit: externalGasLimit,
         });
-    }
 
-    async walletFactoryEstimateGasOfDeployCall(
-        request: DeployTransactionRequest
-    ): Promise<number> {
-        if (
-            request.metadata.relayHubAddress === undefined ||
-            request.metadata.relayHubAddress === null ||
-            request.metadata.relayHubAddress === constants.ZERO_ADDRESS
-        ) {
-            throw new Error('calculateDeployCallGas: RelayHub must be defined');
-        }
-        const rHub = await this._createRelayHub(
-            request.metadata.relayHubAddress
-        );
-        const method = rHub.contract.methods.deployCall(
-            request.relayRequest,
-            request.metadata.signature
+      return {
+        verifierAccepted: true,
+        reverted: false,
+        returnValue: res.hash, //res.returnValue was the original value, but I have no idea what it is meant to be? the deployCall method does not return anything.
+      };
+    } catch ({ message }) {
+      return {
+        verifierAccepted: true,
+        reverted: true,
+        returnValue: `view call to 'deployCall' reverted in client: ${
+          message as string
+        }`,
+      };
+    }
+  }
+
+  async getMaxViewableDeployGasLimit(
+    request: DeployTransactionRequest
+  ): Promise<BigNumberish> {
+    const { relayRequest } = request;
+    const {
+      relayData: { gasPrice, relayWorker },
+    } = relayRequest;
+
+    if (BigNumber.from(gasPrice).eq(0)) {
+      return 0;
+    }
+
+    const maxEstimatedGas = await this.walletFactoryEstimateGasOfDeployCall(
+      request
+    );
+    const workerBalanceAsUnitsOfGas = (
+      await this.getBalance(relayWorker as string)
+    ).div(BigNumber.from(gasPrice));
+
+    return workerBalanceAsUnitsOfGas.gte(maxEstimatedGas) ? maxEstimatedGas : 0;
+  }
+
+  async estimateRelayTransactionMaxPossibleGas(
+    relayRequest: EnvelopingTypes.RelayRequestStruct,
+    signature: string
+  ): Promise<BigNumber> {
+    const encodedTargetCall = this._relayHub.interface.encodeFunctionData(
+      'relayCall',
+      [relayRequest, signature]
+    );
+    const maxPossibleGas = await this.provider.estimateGas({
+      from: relayRequest.relayData.relayWorker,
```

Comment:
now we have a problem, not having the relayWorker address as the from does not necessarily imply fees receiver but certainly requires to be the relay worker right @rsksmart/payments ?

Reply:
correct, at this point the RelayWorker needs to be pass in the function parameter or be sure that is available where we needed in that class.